### PR TITLE
CT-2929 improve feature spec coverage

### PR DESCRIPTION
**Pull request by willmcb**
## Description
Refactors and adds to the Offender SAR case `case_creation_spec.rb` to reflect the four "happy path" journeys for creating Offender SARs. These are:

Screenshot 2020-07-20 at 11 03 54

## Notes for reviewer
- I haven't changed any off the expectations for the case summary page at the end of the journey. I am wondering if this should be covered by  `rspec/views/cases/offender_sar/_case_details_html_slim_spec.rb` would be good to get your opinion on this.

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [x] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
- N/A tests only

### Related JIRA tickets
[CT-2929](https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2929)

### Deployment
- N/A

### Manual testing instructions
- go through each scenario as above
- alternatively add some `sleep(5)` into the methods in the spec and run the following command to use Chrome to inspect the specs visually as the run:

`CHROME_DEBUG=1 bundle exec rspec spec/features/cases/offender_sar/case_creating_spec.rb`

### CODE REVIEW

File: spec/features/cases/offender_sar/case_creating_spec.rb [hunk 1]
```diff
@@ -10,45 +10,109 @@
     cases_page.load
   end
 
-  scenario 'creating a case that does not need clearance', js: true do
+  scenario '1 Third party requests data to be sent to the data subject' do
```
Comment (Polsonby) [inline spec/features/cases/offender_sar/case_creating_spec.rb:13]: This is great - I like the cucumber style method names - it's not in the same style as the rest of the features specs here but happy to adopt this style 

Just wondering if the scenario names properly reflect what's happening? 

I think the `and_fill_in_requester_details_page` method defaults to subject requesting the data

**Comment by willmcb** [comment]
Crib sheet of scenarios:

File: spec/features/cases/offender_sar/case_creating_spec.rb (hunk 1 above)
Comment (willmcb) [reply spec/features/cases/offender_sar/case_creating_spec.rb]: Have updated the spec to reflect this 👍

**Review by Polsonby — APPROVED** [review]
Great job :)

**Pull request merged**